src/MizarData/MizarData.cpp

```diff
@@ -29,10 +38,86 @@ void MizarData::OnTimer(const orbit_client_protos::TimerInfo& timer_info) {
 }
 
 std::optional<std::string> MizarData::GetFunctionNameFromAddress(uint64_t address) const {
-  const orbit_client_data::LinuxAddressInfo* address_info =
-      GetCaptureData().GetAddressInfo(address);
-  if (address_info == nullptr) return std::nullopt;
-  return address_info->function_name();
+  const std::string name =
+      orbit_client_data::GetFunctionNameByAddress(*module_manager_, GetCaptureData(), address);
+
+  if (name == orbit_client_data::kUnknownFunctionOrModuleName) return std::nullopt;
+  return name;
+}
+
+void MizarData::UpdateModules(const std::vector<orbit_grpc_protos::ModuleInfo>& module_infos) {
+  for (const auto* not_updated_module :
+       module_manager_->AddOrUpdateNotLoadedModules(module_infos)) {
+    ORBIT_LOG("Module %s is not updated", not_updated_module->file_path());
+  }
+  GetMutableCaptureData().mutable_process()->UpdateModuleInfos(module_infos);
+}
+
+void MizarData::LoadSymbolsForAllModules() {
+  for (const orbit_client_data::ModuleData* module_data : module_manager_->GetAllModuleData()) {
+    orbit_client_data::ModuleData* mutable_module_data =
+        module_manager_->GetMutableModuleByPathAndBuildId(module_data->file_path(),
+                                                          module_data->build_id());
+    LoadSymbols(*mutable_module_data);
+  }
+}
+
+static ErrorMessageOr<std::filesystem::path> SearchSymbolsPathInOrbitSearchPaths(
+    const orbit_symbols::SymbolHelper& symbol_helper,
+    const orbit_client_data::ModuleData& module_data) {
+  // These are the constants used by Orbit Client. This way we read its configs.
+  static const QString orbit_organization = QStringLiteral("The Orbit Authors");
+  static const QString orbit_app_name = QStringLiteral("orbitprofiler");
+  orbit_client_symbols::QSettingsBasedStorageManager storage_manager(orbit_organization,
+                                                                     orbit_app_name);
+  return symbol_helper.FindSymbolsFileLocally(module_data.file_path(), module_data.build_id(),
+                                              module_data.object_file_type(),
+                                              storage_manager.LoadPaths());
+}
+
+static void LogSymbolsFound(std::string_view module_path, std::string_view symbols_path) {
+  ORBIT_LOG("Found symbol path for module \"%s\". Symbols filename: \"%s\"", module_path,
+            symbols_path);
+}
+
+static ErrorMessageOr<std::filesystem::path> FindSymbolsPath(
+    const orbit_symbols::SymbolHelper& symbol_helper,
+    const orbit_client_data::ModuleData& module_data) {
+  if (auto symbols_paths_or_error = SearchSymbolsPathInOrbitSearchPaths(symbol_helper, module_data);
+      symbols_paths_or_error.has_value()) {
+    LogSymbolsFound(module_data.file_path(), symbols_paths_or_error.value().string());
+    return symbols_paths_or_error;
+  }
+
+  OUTCOME_TRY(auto symbols_paths,
+              symbol_helper.FindSymbolsInCache(module_data.file_path(), module_data.file_size()));
```

Comment:
You are now only checking the cache for files with the same file size, aka the situation when symbols where included in the module file itself. You should also call the other overload, that checks for a file in cache with the same build id or you are missing files.

Reply:
I think this justifies a comment as well because in the general case, the symbols file is not the same as the module file.

Reply:
@beckerhe yep.